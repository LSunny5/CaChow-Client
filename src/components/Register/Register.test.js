import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationForm from './Register';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`RegistrationForm route`, () => {
    const props = {
        error: {}, 
        onRegistrationSuccess: () => {}, 
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <RegistrationForm {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the RegistrationForm form', () => {
        const wrapper = shallow(<RegistrationForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});