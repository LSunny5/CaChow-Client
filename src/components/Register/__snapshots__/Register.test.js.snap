// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm route renders the RegistrationForm form 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <legend
    className="formTitle"
  >
    Register
  </legend>
  <div
    role="alert"
  />
  <div
    className="fullName"
  >
    <label
      htmlFor="fullNameInput"
    >
      Full name 
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      id="fullNameInput"
      name="full_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="userNameInput"
    >
      User name 
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      id="userNameInput"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="passwordInput"
    >
      Password 
      <span
        className="required"
      >
        *
      </span>
    </label>
    <input
      id="passwordInput"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <div
      className="buttonBox"
    >
      <button
        className="borderButton"
        type="submit"
      >
        Register
      </button>
    </div>
  </div>
</form>
`;
